### DISCUSSION TITLE
Support some configuration (i.e. price plan) for the container app environment

### DISCUSSION THREAD

**Discussion by mip1983** [microsoft/aspire#4337] [Ideas]
Fairly recently in azd 1.9, the container app environment that get's generated from an aspire project has gone from the consumption only plan to a workload profile. It can't change plan if you've got something pre-1.9 so this breaks unless you use infra synth to change it back. Going to this plan would introduce a fixed monthly cost and I'd want to stay on consumption only (particularly on the dev environment, but all environments really for the moment).

Are there any plans to introduce some control into aspire to be able to configure the container apps environment, particularly the price plan, without needing infra-synth? It's a parameter you'd want to be able to change on a deployment environment basis, so for it to come from Aspire I guess it would need to come from app config or environment variable.

I raised it here on the azure dev repo as well, as I'm not sure what side it best sits. It could arguably be an up-front input into azd like subscription id etc. is at the moment.

https://github.com/Azure/azure-dev/discussions/3874#discussioncomment-9357469

**Comment by davidfowl** [comment]
Yes, some big architectural changes required to change *anything* about the container app and its something we didn't get to for GA. The plan is to tackle it like how we handle the rest of the azure resources so it can be configured in the apphost. It's not coming in the next release though, there are some bigger changes that need to happen to make this work. You can follow this work here https://github.com/Azure/azure-dev/issues/3292.

Until then infra synth is your only way out.

**Comment by SalomonHenke** [comment]
infra synth ? just so i understand it correctly, it means i cant pre-configure ingress or cors or similar? So i have to manually edit it after running the azd deploy?

**Comment by mip1983** [comment]
Thanks @davidfowl, good to know it's in the works and be able to follow along.
